### ISSUE TITLE
Crashes if the custom template is used

### ISSUE THREAD

**Issue by DeadlySquad13** [danymat/neogen#95]
Hi!
I've just installed this package. Everything works nicely until I try to change template settings.

**Expected**:
Normally functioning reST docstrings in python.

**Actually**:
```
E5108: Error executing lua ...m/site/pack/packer/start/neogen/lua/neogen/generator.lua:203: attempt to index field 'parent' (a nil value)
stack traceback:
        ...m/site/pack/packer/start/neogen/lua/neogen/generator.lua:203: in function 'generate'
        [string ":lua"]:1: in main chunk
```

**Configuration**:
neovim version (output of `nvim --version`):
```sh
NVIM v0.8.0-dev+319-g285e73894
Build type: RelWithDebInfo
LuaJIT 2.1.0-beta3
Compilation: /usr/bin/gcc-11 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -DNVIM_TS_HAS_SET_MATCH_LIMIT -DNVIM_TS_HAS_SET_ALLOCATOR -O2 -g -Og -g -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wdouble-promotion -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-prototypes -Wimplicit-fallthrough -Wsuggest-attribute=pure -Wsuggest-attribute=const -Wsuggest-attribute=malloc -Wsuggest-attribute=cold -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -DMIN_LOG_LEVEL=3 -I/home/runner/work/neovim/neovim/build/config -I/home/runner/work/neovim/neovim/src -I/home/runner/work/neovim/neovim/.deps/usr/include -I/usr/include -I/home/runner/work/neovim/neovim/build/src/nvim/auto -I/home/runner/work/neovim/neovim/build/include
Compiled by runner@fv-az294-755

Features: +acl +iconv +tui
See ":help feature-compile"

   system vimrc file: "$VIM/sysinit.vim"
  fall-back for $VIM: "
/home/runner/work/neovim/neovim/build/nvim.AppDir/usr/share/nvim"

Run :checkhealth for more info
```

plugins.lua (part regarding neogen):
```lua
use({
  'danymat/neogen',
  config = [[ require('config.neogen') ]],
  requires = 'nvim-treesitter/nvim-treesitter',
});
```

config/neogen/init.lua:
```lua
local status_ok, neogen = pcall(require, 'neogen')
if not status_ok then
  return
end

neogen.setup({
  enabled = true,
  input_after_comment = true, -- (default: true) automatic jump (with insert mode) on inserted annotation
  languages = {
    python = {
      template = {
        annotation_convention = 'reST'
      }
    }
  }
})
```

**My small investigation**:
- Tried to change to stable version of the package, didn't fix the problem.
- Checked similar issues, found the one regarding [c# not working](https://github.com/danymat/neogen/issues/62).
  There was a problem with treesitter namings. Fortunately, it's fixed but I
  have the same error anyway.
- After looking into c# issue, I decided to check treesitter. According to
  checkhealth, everything is ok:
```sh
nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.6 (ccd6bf554d922596ce905730d98a77af368bba5c) (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v14.18.3 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `gcc` executable found. Selected from { "gcc", "clang" }
    Version: bash: cannot set terminal process group (-1): Inappropriate ioctl for device
  - OK: Neovim was compiled with tree-sitter runtime ABI version 14 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - bash           ✓ ✓ ✓ . ✓ 
  - javascript     ✓ ✓ ✓ ✓ ✓ 
  - lua            ✓ ✓ ✓ ✓ ✓ 
  - cpp            ✓ ✓ ✓ ✓ ✓ 
  - scss           ✓ . . ✓ . 
  - python         ✓ ✓ ✓ ✓ ✓ 
  - regex          ✓ . . . . 
  - c              ✓ ✓ ✓ ✓ ✓ 
  - typescript     ✓ ✓ ✓ ✓ ✓ 
  - cmake          ✓ . ✓ . . 
  - css            ✓ . ✓ ✓ ✓ 
  - tsx            ✓ ✓ ✓ ✓ ✓ 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}
```

- Thought, maybe have some problem with template configuration, left just
  python, error persists:
```lua
neogen.setup({
  enabled = true,
  input_after_comment = true, -- (default: true) automatic jump (with insert mode) on inserted annotation
  languages = {
    python = {
      --template = {
        --annotation_convention = 'reST'
      --}
    }
  }
})
```

**Workaround**:
Just hardcode `reST` into `default_annotation` in the configuration file.
Of course, it's poor solution, I hope anyone helps to solve this issue!

**Comment by danymat** [comment]
Hello, I hope you're doing okay ! 

I tried to reproduce your issue by putting:

```lua
		use({
			-- "danymat/neogen",
			"~/Developer/neogen/",
			config = function()
				-- require("neogen").setup({ snippet_engine = "luasnip" })
				require("neogen").setup({
					enabled = true,
					input_after_comment = true, -- (default: true) automatic jump (with insert mode) on inserted annotation
					languages = {
						python = {
							template = {
								annotation_convention = "reST",
							},
						},
					},
				})
			end,
			requires = "nvim-treesitter/nvim-treesitter",
		})
```

And everything works on my end.

Can you try again ? I'm not sure where is the bug here.
To help me reproduce:

- Do not forget to PackerCompile again
- Add a minimal python file where the bug happens

**Comment by DeadlySquad13** [comment]
> 

Hello,
Thanks for the attention to my issue! I tried your config: everything worked properly. I tried again with my config file - I have tested two annotations just to be sure (reST and numpydoc) - it was fine too. The issue can probably be closed.

Regarding reproduce:
- I was testing it on multiple python files from my personal projects, can't make it minimal as I don't have bug now.
- I do PackerCompile every time I change the plugins. While I was trying to fix neogen, I was reinstalling it multiple times jumping from stable version to latest: I should have compiled during it at least once.

If you still want to investigate it, I will try to help.
I have seen the line `-- require("neogen").setup({ snippet_engine = "luasnip" })` in your config. During the last 3 days I was fixing UltiSnips + nvim-cmp integration, can it affect this plugin somehow? Also I'm now on the latest treesitter version (commit SHA `5c364707442965ae52a1ac89c359080e32958445`), don't know if they fixed something that was causing an issue.
